ergo-wallet/src/main/scala/org/ergoplatform/wallet/boxes/TrackedBox.scala

```diff
@@ -92,8 +97,17 @@ object TrackedBoxSerializer extends ErgoWalletSerializer[TrackedBox] {
     w.putOption(obj.inclusionHeightOpt)(_.putInt(_))
     w.putOption(obj.spendingTxIdOpt)((bf, id) => bf.putBytes(idToBytes(id)))
     w.putOption(obj.spendingHeightOpt)(_.putInt(_))
-    w.put(if (obj.certainty.certain) 0x01 else 0x00)
-    w.putShort(obj.applicationId)
+
+    val appsCount = obj.applicationStatuses.size.toShort
+
+    if (appsCount == 1 && obj.applicationStatuses.head == Constants.PaymentsAppId) {
+      w.putShort(0)
+    } else {
+      w.putShort(appsCount)
+      obj.applicationStatuses.foreach { appId =>
+        w.putShort(appId)
+      }
+    }
```

Comment:
A comments related to many places, not only this one: 
1) it is better to always and consistently use unsigned versions of methods (putUxxx) in all cases where negative values doesn't make sense. (performance and clarity)
2) In addition, conversions for unsigned to signed should use Exact versions (like toIntExact).
Please review all serializers as I saw similar issues in many cases.

Reply:
oh, for application ids maybe better to use signed shorts, to have reserve in negative space (not sure whether it will be needed and when though).